### ISSUE TITLE
DataTable/TreeTable rowClassName/cellClassName API redundancy

### ISSUE THREAD

**Issue by VsevolodGolovanov** [primefaces/primereact#2520]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**

**Current behavior**
Current API forces you to return a weird object with a condition
```
rowClassName?(data: any): object;
cellClassName?(data: any, options: any): object;
```
> returns an object in "{'styleclass' : condition}" format to define a classname

This seems completely unnecessary and convoluted for rowClassName, since it gets called for each row anyway with data = row. Same with cellClassName - it's called for each cell.

**Expected behavior**
rowClassName could look like:
```
rowClassName?(rowData: any): string;
```
So rowClass from https://www.primefaces.org/primereact/datatable/style/ would become:
```
rowClass(rowData) {
	return rowData.category === 'Accessories' ?  'row-accessories' : '';
}
```
Same for cellClassName. Except I'd replace the fat options argument with a more specific signature:
```
cellClassName?(rowData: any, cellData: any, field: string): string;
```
Why is cellData not enough? Because in my case the rowData object is a complex metadata driven class instance that I need to call methods on. So my field is just a name of a property described by metadata, not a literally resolvable `rowData[field]`. I have a custom bodyTemplate, sortFunction etc  that define how to get the actual cell values or textual representations, what to do with them.

**Minimal reproduction of the problem with instructions**

**Please tell us about your environment:**

* **React version:**

* **PrimeReact version:**
6.5.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by VsevolodGolovanov** [comment]
Looks like integrated https://www.npmjs.com/package/classnames functionality, but a developer can easily use it themselves in their callback if they want to.

**Comment by melloware** [comment]
Do you want to submit a proposed PR?

**Comment by melloware** [comment]
These were updated the new signatures are now this as of 8.0.0

```ts
    rowClassName?(data: any, options: DataTableRowClassNameOptions): object | string;
    cellClassName?(value: any, options: DataTableCellClassNameOptions): object | string;
```

**Comment by melloware** [comment]
SO @VsevolodGolovanov can you verify if this issue can be closed based on 8.0.0 new signature and use?

**Comment by VsevolodGolovanov** [comment]
Yep, seems fine.
